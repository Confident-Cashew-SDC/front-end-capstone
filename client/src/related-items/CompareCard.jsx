import React from 'react';
import CompareCSS from './cssModules/Compare.module.css';

const CompareCard = ({ itemName, itemFeatures, parentItemName, parentFeatures }) => {
  return (
    <>
      <div id = 'compare-card' className={CompareCSS.item}>
        <h3 className={CompareCSS.h3}>{itemName}</h3>
        {itemFeatures.map((element) => (
          <ul className={CompareCSS.list}>
            <li>{element.feature} - {element.value}</li>
          </ul>
        ))}
      </div>
      <div className={CompareCSS.item}>
        <h3 className={CompareCSS.h3}>{parentItemName}</h3>
        {parentFeatures.map((element) => (
          <ul className={CompareCSS.list}>
            <li>{element.feature} - {element.value}</li>
          </ul>
        ))}
      </div>
    </>
  )
};
export default CompareCard;