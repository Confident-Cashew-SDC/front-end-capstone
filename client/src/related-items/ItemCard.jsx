import React from 'react';

//import Item from './Item.jsx';
import StarRating from './StarRating.jsx';
import ItemCSS from './cssModules/Item.module.css';
import RelatedCSS from './cssModules/Related.module.css';

const ItemCard = ({ category, image, name, noImage, id, parentId, price, clickOnTitle, compareButton}) => {
  return (
    <>
      <div id = 'card' className={ItemCSS.inner}>
        <h4 className={ItemCSS.h4}>{category}</h4>
        <h2 className={ItemCSS.item_title} onClick={clickOnTitle}>{name}</h2>
        {image === undefined
          ?
          <div className={RelatedCSS.loader_container}>
            <div className={RelatedCSS.loader}></div>
          </div>
          :
          <img className={ItemCSS.image} src={image === null ? noImage : image} />
        }
        <p className={ItemCSS.par}>{`$${price}`}</p>
        <StarRating
          id={id}
          parentId={parentId}
        />
          <button className={ItemCSS.button_compare} onClick={compareButton}>Compare Me</button>
      </div>
    </>
  )
}

export default ItemCard;


