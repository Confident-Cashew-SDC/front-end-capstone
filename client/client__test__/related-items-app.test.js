import React from 'react';
import { render, fireEvent, waitForElement, screen } from '@testing-library/react';
import AppRelated from '../src/related-items/related-items-app.jsx';
import ItemCard from '../src/related-items/ItemCard.jsx';
import CompareCard from '../src/related-items/CompareCard.jsx';

const product_id = 63609;

const parentItemStub = {
  "id": 63609,
  "campus": "hr-sfo",
  "name": "Camo Onesie",
  "slogan": "Blend in to your crowd",
  "description": "The So Fatigues will wake you up and fit you in. This high energy camo will have you blending in to even the wildest surroundings.",
  "category": "Jackets",
  "default_price": "140.00",
  "created_at": "2021-12-21T17:19:40.556Z",
  "updated_at": "2021-12-21T17:19:40.556Z",
  "features": [
    {
      "feature": "Fabric",
      "value": "Canvas"
    },
    {
      "feature": "Buttons",
      "value": "Brass"
    }
  ]
}

// items related to parent item
const relatedArray = [
  63610,
  63611,
  63616,
  63615
];

const thumbnailplaceHolder = 'https://www.budget101.com/images/image-not-available.png?14867';


test('should render with a product id passed as a prop and see related items as a title',async () => {
  await render(<AppRelated product_id={product_id} />);
  expect(screen.getByText("FETCHING RELATED ITEMS"));
});

const stubbedItemOne = {
  'parentId': "63609",
  'id': 63610,
  'category': 'Accessories',
  'image': null,
  'price': '69.00',
  'name': 'Bright Future Sunglasses'
};


const stubbedItemTwo = {
  'parentId': "63609",
  'id': 63611,
  'category': 'Pants',
  'image': 'https://images.unsplash.com/photo-1552902865-b72c031ac5ea?ixlib=rb-1.2.1&ixid=eyJhcHBfaWQiOjEyMDd9&auto=format&fit=crop&w=300&q=80',
  'price': '40.00',
  'name': 'Morning Joggers'
}


const stubbedItemThree = {
  'parentId': "63609",
  'id': 63616,
  'category': 'Kicks',
  'image': 'https://images.unsplash.com/photo-1551489186-cf8726f514f8?ixlib=rb-1.2.1&ixid=eyJhcHBfaWQiOjEyMDd9&auto=format&fit=crop&w=300&q=80',
  'price': '450.00',
  'name': 'YEasy 350'
}


const stubbedItemFour = {
  'parentId': "63609",
  'id': 63615,
  'category': 'Dress Shoes',
  'image': 'https://images.unsplash.com/photo-1561861422-a549073e547a?ixlib=rb-1.2.1&ixid=eyJhcHBfaWQiOjEyMDd9&auto=format&fit=crop&w=300&q=80',
  'price': '120.00',
  'name': 'Blues Suede Shoes'
}

it('matches snapshot of Item One', async () => {
  const { asFragment } = await render(<ItemCard id={stubbedItemOne.id} parentId={stubbedItemOne.parentId} category={stubbedItemOne.category} name={stubbedItemOne.name} image={stubbedItemOne.image} price={stubbedItemOne.price} noImage={thumbnailplaceHolder} />);
  expect(asFragment()).toMatchSnapshot();
});


it('matches snapshot of Item Two', async () => {
  const { asFragment } = await render(<ItemCard id={stubbedItemTwo.id} parentId={stubbedItemTwo.parentId} category={stubbedItemTwo.category} name={stubbedItemTwo.name} image={stubbedItemTwo.image} price={stubbedItemTwo.price} noImage={thumbnailplaceHolder} />);
  expect(asFragment()).toMatchSnapshot();
});



it('matches snapshot of Item Three', async () => {
  const { asFragment } = await render(<ItemCard id={stubbedItemThree.id} parentId={stubbedItemThree.parentId} category={stubbedItemThree.category} name={stubbedItemThree.name} image={stubbedItemThree.image} price={stubbedItemThree.price} noImage={thumbnailplaceHolder} />);
  expect(asFragment()).toMatchSnapshot();
});



it('matches snapshot of Item Four', async () => {
  const { asFragment } = await render(<ItemCard id={stubbedItemFour.id} parentId={stubbedItemFour.parentId} category={stubbedItemFour.category} name={stubbedItemFour.name} image={stubbedItemFour.image} price={stubbedItemFour.price} noImage={thumbnailplaceHolder} />);
  expect(asFragment()).toMatchSnapshot();
});


//each card needs to have a compare me button
it('each card has compare button', () => {
  render(<ItemCard id={relatedArray[0]} />);
  expect(screen.getByText("Compare Me"));
});


//snapshot tests for compare card modal

//id = 63610
const itemOneFeaturesStub = [
  { feature: 'Lenses', value: 'Ultrasheen' },
  { feature: 'UV Protection', value: null },
  { feature: 'Frames', value: 'LightCompose' }
];


//id = 63611
const itemTwoFeaturesStub = [
  { feature: 'Fabric', value: '100% Cotton' },
  { feature: 'Cut', value: 'Skinny' }
]

//id = 63616
const itemThreeFeaturesStub = [
  { feature: 'Sole', value: 'Rubber' },
  { feature: 'Material', value: 'FullControlSkin' },
  { feature: 'Stitching', value: 'Double Stitch' }
]

//id = 63615
const itemFourFeaturesStub = [
  { feature: 'Sole', value: 'Rubber' },
  { feature: 'Material', value: 'FullControlSkin' },
  { feature: 'Stitching', value: 'Double Stitch' }
]

it('matches snapshot of features for Item One', async () => {
  const { asFragment } = await render(<CompareCard id={stubbedItemOne.id} parentId={parentItemStub.id} parentFeatures={parentItemStub.features} itemFeatures={itemOneFeaturesStub} itemName={stubbedItemOne.name} parentItemName={parentItemStub.name} />);
  expect(asFragment()).toMatchSnapshot();
});


it('matches snapshot of features for Item Two', async () => {
  const { asFragment } = await render(<CompareCard id={stubbedItemTwo.id} parentId={parentItemStub.id} parentFeatures={parentItemStub.features} itemFeatures={itemTwoFeaturesStub} itemName={stubbedItemTwo.name} parentItemName={parentItemStub.name} />);
  expect(asFragment()).toMatchSnapshot();
});


it('matches snapshot of features for Item Three', async () => {
  const { asFragment } = await render(<CompareCard id={stubbedItemThree.id} parentId={parentItemStub.id} parentFeatures={parentItemStub.features} itemFeatures={itemThreeFeaturesStub} itemName={stubbedItemThree.name} parentItemName={parentItemStub.name} />);
  expect(asFragment()).toMatchSnapshot();
});


it('matches snapshot of features for Item Four', async () => {
  const { asFragment } = await render(<CompareCard id={stubbedItemFour.id} parentId={parentItemStub.id} parentFeatures={parentItemStub.features} itemFeatures={itemFourFeaturesStub} itemName={stubbedItemFour.name} parentItemName={parentItemStub.name} />);
  expect(asFragment()).toMatchSnapshot();
});


