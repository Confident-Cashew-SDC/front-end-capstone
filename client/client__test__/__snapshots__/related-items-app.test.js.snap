// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div
    id="card"
  >
    <h4>
      Accessories
    </h4>
    <h2>
      Bright Future Sunglasses
    </h2>
    <img
      src="https://www.budget101.com/images/image-not-available.png?14867"
    />
    <p>
      $69.00
    </p>
    <span>
      <span>
        <h4>
          Average rating: 
        </h4>
        <div>
          <svg
            height="0"
            width="0"
            x="0"
            xmlns="http://www.w3.org/2000/svg"
            y="0"
          >
            <defs>
              <mask
                id="mask030"
              >
                <rect
                  fill="white"
                  height="30"
                  viewBox="0 0 24 24"
                  width="30"
                  x="0"
                  y="0"
                />
                <rect
                  fill="black"
                  height="30"
                  viewBox="0 0 24 24"
                  width="30"
                  x="0"
                  y="0"
                />
              </mask>
              <symbol
                id="star"
                viewBox="0 0 24 24"
                xlmns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M12 .587l3.668 7.568 8.332 1.151-6.064 5.828 1.48 8.279-7.416-3.967-7.417 3.967 1.481-8.279-6.064-5.828 8.332-1.151z"
                />
              </symbol>
            </defs>
          </svg>
          <div>
            <svg
              height="30"
              viewBox="0 0 24 24"
              width="30"
            >
              <use
                fill="none"
                href="#star"
                stroke="black"
              />
            </svg>
            <svg
              height="30"
              viewBox="0 0 24 24"
              width="30"
            >
              <use
                fill="none"
                href="#star"
                stroke="black"
              />
            </svg>
            <svg
              height="30"
              viewBox="0 0 24 24"
              width="30"
            >
              <use
                fill="none"
                href="#star"
                stroke="black"
              />
            </svg>
            <svg
              height="30"
              viewBox="0 0 24 24"
              width="30"
            >
              <use
                fill="none"
                href="#star"
                stroke="black"
              />
            </svg>
            <svg
              height="30"
              viewBox="0 0 24 24"
              width="30"
            >
              <use
                fill="none"
                href="#star"
                stroke="black"
              />
            </svg>
          </div>
        </div>
      </span>
    </span>
    <button>
      Compare Me
    </button>
  </div>
</DocumentFragment>
`;

exports[`matches snapshot 2`] = `
<DocumentFragment>
  <div
    id="card"
  >
    <h4>
      Pants
    </h4>
    <h2>
      Morning Joggers
    </h2>
    <img
      src="https://images.unsplash.com/photo-1552902865-b72c031ac5ea?ixlib=rb-1.2.1&ixid=eyJhcHBfaWQiOjEyMDd9&auto=format&fit=crop&w=300&q=80"
    />
    <p>
      $40.00
    </p>
    <span>
      <span>
        <h4>
          Average rating: 
        </h4>
        <div>
          <svg
            height="0"
            width="0"
            x="0"
            xmlns="http://www.w3.org/2000/svg"
            y="0"
          >
            <defs>
              <mask
                id="mask030"
              >
                <rect
                  fill="white"
                  height="30"
                  viewBox="0 0 24 24"
                  width="30"
                  x="0"
                  y="0"
                />
                <rect
                  fill="black"
                  height="30"
                  viewBox="0 0 24 24"
                  width="30"
                  x="0"
                  y="0"
                />
              </mask>
              <symbol
                id="star"
                viewBox="0 0 24 24"
                xlmns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M12 .587l3.668 7.568 8.332 1.151-6.064 5.828 1.48 8.279-7.416-3.967-7.417 3.967 1.481-8.279-6.064-5.828 8.332-1.151z"
                />
              </symbol>
            </defs>
          </svg>
          <div>
            <svg
              height="30"
              viewBox="0 0 24 24"
              width="30"
            >
              <use
                fill="none"
                href="#star"
                stroke="black"
              />
            </svg>
            <svg
              height="30"
              viewBox="0 0 24 24"
              width="30"
            >
              <use
                fill="none"
                href="#star"
                stroke="black"
              />
            </svg>
            <svg
              height="30"
              viewBox="0 0 24 24"
              width="30"
            >
              <use
                fill="none"
                href="#star"
                stroke="black"
              />
            </svg>
            <svg
              height="30"
              viewBox="0 0 24 24"
              width="30"
            >
              <use
                fill="none"
                href="#star"
                stroke="black"
              />
            </svg>
            <svg
              height="30"
              viewBox="0 0 24 24"
              width="30"
            >
              <use
                fill="none"
                href="#star"
                stroke="black"
              />
            </svg>
          </div>
        </div>
      </span>
    </span>
    <button>
      Compare Me
    </button>
  </div>
</DocumentFragment>
`;

exports[`matches snapshot of Item Four 1`] = `
<DocumentFragment>
  <div
    id="card"
  >
    <h4>
      Dress Shoes
    </h4>
    <h2>
      Blues Suede Shoes
    </h2>
    <img
      src="https://images.unsplash.com/photo-1561861422-a549073e547a?ixlib=rb-1.2.1&ixid=eyJhcHBfaWQiOjEyMDd9&auto=format&fit=crop&w=300&q=80"
    />
    <p>
      $120.00
    </p>
    <span>
      <span>
        <h4>
          Average rating: 
        </h4>
        <div>
          <svg
            height="0"
            width="0"
            x="0"
            xmlns="http://www.w3.org/2000/svg"
            y="0"
          >
            <defs>
              <mask
                id="mask030"
              >
                <rect
                  fill="white"
                  height="30"
                  viewBox="0 0 24 24"
                  width="30"
                  x="0"
                  y="0"
                />
                <rect
                  fill="black"
                  height="30"
                  viewBox="0 0 24 24"
                  width="30"
                  x="0"
                  y="0"
                />
              </mask>
              <symbol
                id="star"
                viewBox="0 0 24 24"
                xlmns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M12 .587l3.668 7.568 8.332 1.151-6.064 5.828 1.48 8.279-7.416-3.967-7.417 3.967 1.481-8.279-6.064-5.828 8.332-1.151z"
                />
              </symbol>
            </defs>
          </svg>
          <div>
            <svg
              height="30"
              viewBox="0 0 24 24"
              width="30"
            >
              <use
                fill="none"
                href="#star"
                stroke="black"
              />
            </svg>
            <svg
              height="30"
              viewBox="0 0 24 24"
              width="30"
            >
              <use
                fill="none"
                href="#star"
                stroke="black"
              />
            </svg>
            <svg
              height="30"
              viewBox="0 0 24 24"
              width="30"
            >
              <use
                fill="none"
                href="#star"
                stroke="black"
              />
            </svg>
            <svg
              height="30"
              viewBox="0 0 24 24"
              width="30"
            >
              <use
                fill="none"
                href="#star"
                stroke="black"
              />
            </svg>
            <svg
              height="30"
              viewBox="0 0 24 24"
              width="30"
            >
              <use
                fill="none"
                href="#star"
                stroke="black"
              />
            </svg>
          </div>
        </div>
      </span>
    </span>
    <button>
      Compare Me
    </button>
  </div>
</DocumentFragment>
`;

exports[`matches snapshot of Item One 1`] = `
<DocumentFragment>
  <div
    id="card"
  >
    <h4>
      Accessories
    </h4>
    <h2>
      Bright Future Sunglasses
    </h2>
    <img
      src="https://www.budget101.com/images/image-not-available.png?14867"
    />
    <p>
      $69.00
    </p>
    <span>
      <span>
        <h4>
          Average rating: 
        </h4>
        <div>
          <svg
            height="0"
            width="0"
            x="0"
            xmlns="http://www.w3.org/2000/svg"
            y="0"
          >
            <defs>
              <mask
                id="mask030"
              >
                <rect
                  fill="white"
                  height="30"
                  viewBox="0 0 24 24"
                  width="30"
                  x="0"
                  y="0"
                />
                <rect
                  fill="black"
                  height="30"
                  viewBox="0 0 24 24"
                  width="30"
                  x="0"
                  y="0"
                />
              </mask>
              <symbol
                id="star"
                viewBox="0 0 24 24"
                xlmns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M12 .587l3.668 7.568 8.332 1.151-6.064 5.828 1.48 8.279-7.416-3.967-7.417 3.967 1.481-8.279-6.064-5.828 8.332-1.151z"
                />
              </symbol>
            </defs>
          </svg>
          <div>
            <svg
              height="30"
              viewBox="0 0 24 24"
              width="30"
            >
              <use
                fill="none"
                href="#star"
                stroke="black"
              />
            </svg>
            <svg
              height="30"
              viewBox="0 0 24 24"
              width="30"
            >
              <use
                fill="none"
                href="#star"
                stroke="black"
              />
            </svg>
            <svg
              height="30"
              viewBox="0 0 24 24"
              width="30"
            >
              <use
                fill="none"
                href="#star"
                stroke="black"
              />
            </svg>
            <svg
              height="30"
              viewBox="0 0 24 24"
              width="30"
            >
              <use
                fill="none"
                href="#star"
                stroke="black"
              />
            </svg>
            <svg
              height="30"
              viewBox="0 0 24 24"
              width="30"
            >
              <use
                fill="none"
                href="#star"
                stroke="black"
              />
            </svg>
          </div>
        </div>
      </span>
    </span>
    <button>
      Compare Me
    </button>
  </div>
</DocumentFragment>
`;

exports[`matches snapshot of Item Three 1`] = `
<DocumentFragment>
  <div
    id="card"
  >
    <h4>
      Kicks
    </h4>
    <h2>
      YEasy 350
    </h2>
    <img
      src="https://images.unsplash.com/photo-1551489186-cf8726f514f8?ixlib=rb-1.2.1&ixid=eyJhcHBfaWQiOjEyMDd9&auto=format&fit=crop&w=300&q=80"
    />
    <p>
      $450.00
    </p>
    <span>
      <span>
        <h4>
          Average rating: 
        </h4>
        <div>
          <svg
            height="0"
            width="0"
            x="0"
            xmlns="http://www.w3.org/2000/svg"
            y="0"
          >
            <defs>
              <mask
                id="mask030"
              >
                <rect
                  fill="white"
                  height="30"
                  viewBox="0 0 24 24"
                  width="30"
                  x="0"
                  y="0"
                />
                <rect
                  fill="black"
                  height="30"
                  viewBox="0 0 24 24"
                  width="30"
                  x="0"
                  y="0"
                />
              </mask>
              <symbol
                id="star"
                viewBox="0 0 24 24"
                xlmns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M12 .587l3.668 7.568 8.332 1.151-6.064 5.828 1.48 8.279-7.416-3.967-7.417 3.967 1.481-8.279-6.064-5.828 8.332-1.151z"
                />
              </symbol>
            </defs>
          </svg>
          <div>
            <svg
              height="30"
              viewBox="0 0 24 24"
              width="30"
            >
              <use
                fill="none"
                href="#star"
                stroke="black"
              />
            </svg>
            <svg
              height="30"
              viewBox="0 0 24 24"
              width="30"
            >
              <use
                fill="none"
                href="#star"
                stroke="black"
              />
            </svg>
            <svg
              height="30"
              viewBox="0 0 24 24"
              width="30"
            >
              <use
                fill="none"
                href="#star"
                stroke="black"
              />
            </svg>
            <svg
              height="30"
              viewBox="0 0 24 24"
              width="30"
            >
              <use
                fill="none"
                href="#star"
                stroke="black"
              />
            </svg>
            <svg
              height="30"
              viewBox="0 0 24 24"
              width="30"
            >
              <use
                fill="none"
                href="#star"
                stroke="black"
              />
            </svg>
          </div>
        </div>
      </span>
    </span>
    <button>
      Compare Me
    </button>
  </div>
</DocumentFragment>
`;

exports[`matches snapshot of Item Two 1`] = `
<DocumentFragment>
  <div
    id="card"
  >
    <h4>
      Pants
    </h4>
    <h2>
      Morning Joggers
    </h2>
    <img
      src="https://images.unsplash.com/photo-1552902865-b72c031ac5ea?ixlib=rb-1.2.1&ixid=eyJhcHBfaWQiOjEyMDd9&auto=format&fit=crop&w=300&q=80"
    />
    <p>
      $40.00
    </p>
    <span>
      <span>
        <h4>
          Average rating: 
        </h4>
        <div>
          <svg
            height="0"
            width="0"
            x="0"
            xmlns="http://www.w3.org/2000/svg"
            y="0"
          >
            <defs>
              <mask
                id="mask030"
              >
                <rect
                  fill="white"
                  height="30"
                  viewBox="0 0 24 24"
                  width="30"
                  x="0"
                  y="0"
                />
                <rect
                  fill="black"
                  height="30"
                  viewBox="0 0 24 24"
                  width="30"
                  x="0"
                  y="0"
                />
              </mask>
              <symbol
                id="star"
                viewBox="0 0 24 24"
                xlmns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M12 .587l3.668 7.568 8.332 1.151-6.064 5.828 1.48 8.279-7.416-3.967-7.417 3.967 1.481-8.279-6.064-5.828 8.332-1.151z"
                />
              </symbol>
            </defs>
          </svg>
          <div>
            <svg
              height="30"
              viewBox="0 0 24 24"
              width="30"
            >
              <use
                fill="none"
                href="#star"
                stroke="black"
              />
            </svg>
            <svg
              height="30"
              viewBox="0 0 24 24"
              width="30"
            >
              <use
                fill="none"
                href="#star"
                stroke="black"
              />
            </svg>
            <svg
              height="30"
              viewBox="0 0 24 24"
              width="30"
            >
              <use
                fill="none"
                href="#star"
                stroke="black"
              />
            </svg>
            <svg
              height="30"
              viewBox="0 0 24 24"
              width="30"
            >
              <use
                fill="none"
                href="#star"
                stroke="black"
              />
            </svg>
            <svg
              height="30"
              viewBox="0 0 24 24"
              width="30"
            >
              <use
                fill="none"
                href="#star"
                stroke="black"
              />
            </svg>
          </div>
        </div>
      </span>
    </span>
    <button>
      Compare Me
    </button>
  </div>
</DocumentFragment>
`;

exports[`matches snapshot of Item two 1`] = `
<DocumentFragment>
  <div
    id="card"
  >
    <h4>
      Pants
    </h4>
    <h2>
      Morning Joggers
    </h2>
    <img
      src="https://images.unsplash.com/photo-1552902865-b72c031ac5ea?ixlib=rb-1.2.1&ixid=eyJhcHBfaWQiOjEyMDd9&auto=format&fit=crop&w=300&q=80"
    />
    <p>
      $40.00
    </p>
    <span>
      <span>
        <h4>
          Average rating: 
        </h4>
        <div>
          <svg
            height="0"
            width="0"
            x="0"
            xmlns="http://www.w3.org/2000/svg"
            y="0"
          >
            <defs>
              <mask
                id="mask030"
              >
                <rect
                  fill="white"
                  height="30"
                  viewBox="0 0 24 24"
                  width="30"
                  x="0"
                  y="0"
                />
                <rect
                  fill="black"
                  height="30"
                  viewBox="0 0 24 24"
                  width="30"
                  x="0"
                  y="0"
                />
              </mask>
              <symbol
                id="star"
                viewBox="0 0 24 24"
                xlmns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M12 .587l3.668 7.568 8.332 1.151-6.064 5.828 1.48 8.279-7.416-3.967-7.417 3.967 1.481-8.279-6.064-5.828 8.332-1.151z"
                />
              </symbol>
            </defs>
          </svg>
          <div>
            <svg
              height="30"
              viewBox="0 0 24 24"
              width="30"
            >
              <use
                fill="none"
                href="#star"
                stroke="black"
              />
            </svg>
            <svg
              height="30"
              viewBox="0 0 24 24"
              width="30"
            >
              <use
                fill="none"
                href="#star"
                stroke="black"
              />
            </svg>
            <svg
              height="30"
              viewBox="0 0 24 24"
              width="30"
            >
              <use
                fill="none"
                href="#star"
                stroke="black"
              />
            </svg>
            <svg
              height="30"
              viewBox="0 0 24 24"
              width="30"
            >
              <use
                fill="none"
                href="#star"
                stroke="black"
              />
            </svg>
            <svg
              height="30"
              viewBox="0 0 24 24"
              width="30"
            >
              <use
                fill="none"
                href="#star"
                stroke="black"
              />
            </svg>
          </div>
        </div>
      </span>
    </span>
    <button>
      Compare Me
    </button>
  </div>
</DocumentFragment>
`;
